src/Features/LanguageServer/Protocol/Handler/InlayHint/InlayHintResolveHandler.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.InlineHints;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using StreamJsonRpc;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.InlayHint
+{
+    [Method(Methods.InlayHintResolveName)]
+    internal sealed class InlayHintResolveHandler : ILspServiceDocumentRequestHandler<LSP.InlayHint, LSP.InlayHint>
+    {
+        private readonly InlayHintCache _inlayHintCache;
+
+        public InlayHintResolveHandler(InlayHintCache inlayHintCache)
+        {
+            _inlayHintCache = inlayHintCache;
+        }
+
+        public bool MutatesSolutionState => false;
+
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(LSP.InlayHint request)
+            => GetCacheEntry(request).CacheEntry.TextDocumentIdentifier;
+
+        public async Task<LSP.InlayHint> HandleRequestAsync(LSP.InlayHint request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.GetRequiredDocument();
+            var (cacheEntry, memberToResolve) = GetCacheEntry(request);
+
+            var currentSyntaxVersion = await document.GetSyntaxVersionAsync(cancellationToken).ConfigureAwait(false);
+            var cachedSyntaxVersion = cacheEntry.SyntaxVersion;
+
+            if (currentSyntaxVersion != cachedSyntaxVersion)
+            {
+                throw new LocalRpcException($"Cached resolve version {cachedSyntaxVersion} does not match current version {currentSyntaxVersion}")
+                {
+                    ErrorCode = LspErrorCodes.ContentModified
+                };
+            }
+
+            var taggedText = await memberToResolve.GetDescriptionAsync(document, cancellationToken).ConfigureAwait(false);
+
+            request.ToolTip = ProtocolConversions.GetDocumentationMarkupContent(taggedText, document, true);
+            return request;
+        }
+
+        private (InlayHintCache.InlayHintCacheEntry CacheEntry, InlineHint InlineHintToResolve) GetCacheEntry(LSP.InlayHint request)
```

Comment:
since we already have this... why not just defer the text change as well?  it will make our original message much smaller.

Reply:
believe there is a client side bug preventing us from doing that at the moment